provisioning/osx/configure/src/compose-file.js

```diff
@@ -0,0 +1,61 @@
+import execa from 'execa';
+import fs from 'fs';
+import getStream from 'get-stream';
+import path from 'path';
+import R from 'ramda';
+
+export const create = services => {
+  const stackNameAndServices = R.toPairs(R.groupBy(service => service.stack, services));
+
+  const genService = service => `  ${service.name}:
+    ports:
+      - ${service.port}:3000
+`;
+
+  const genStack = services => `version: "3.1"
+
+services:
+${R.join('', R.map(genService, services))}
+`;
+
+  return R.fromPairs(
+    R.map(([stackname, services]) => [stackname, genStack(services)], stackNameAndServices),
+  );
+};
+
+// const deleteFile = ({ name }) => fs.unlinkSync(name);
+
+export const mergeFn = async (cmd, args) => {
+  const cp = execa(cmd, args, {
+    env: {
+      PATH: process.env.PATH,
+    },
+    extendEnv: false,
+  });
+  cp.stderr.pipe(process.stderr);
+  return getStream(cp.stdout);
+};
+
+export const merge = async (mergeFn, dir, filesByStack) => {
+  const retVal = {};
+  for (const [stack, files] of R.toPairs(filesByStack)) {
+    const args = R.chain(f => ['-f', f], R.map(f => path.join(dir, f), files));
+    retVal[stack] = await mergeFn('docker-compose', [...args, 'config']);
+  }
+  return retVal;
+};
```

Comment:
Hmm, not so fast Stupid Stuey. :-)
I tried it on the example below and it doesn't seem to wait for the promise to resolve so the test fails. I looked at the source code for `R.reduce` and there's no handling of async functions in there. :-( Would have been so nice though.